launch_testing/launch_testing/loader.py

```diff
@@ -17,18 +17,57 @@
 import itertools
 import unittest
 
+from .actions import ReadyToTest
+
 
 def _normalize_ld(launch_description_fn):
     # A launch description fn can return just a launch description, or a tuple of
     # (launch_description, test_context).  This wrapper function normalizes things
     # so we always get a tuple, sometimes with an empty dictionary for the test_context
-    def wrapper(*args, **kwargs):
-        result = launch_description_fn(*args, **kwargs)
+    def normalize(result):
         if isinstance(result, tuple):
             return result
         else:
             return result, {}
 
+    def wrapper(**kwargs):
+
+        fn_args = inspect.getfullargspec(launch_description_fn)
+
+        if 'ready_fn' in fn_args.args + fn_args.kwonlyargs:
+            # This is an old-style launch_description function which epects ready_fn to be passed
+            # in to the function
+            return normalize(launch_description_fn(**kwargs))
+        else:
+            # This is a new-style launch_description which should contain a ReadyToTest action
+            ready_fn = kwargs.pop('ready_fn')
+            result = normalize(launch_description_fn(**kwargs))
+            # Fish the ReadyToTest action out of the launch description and plumb our
+            # ready_fn to it
+
+            def iterate_ready_to_test_actions(entities):
+                """Recursively search LaunchDescription entities for all ReadyToTest actions."""
+                for entity in entities:
+                    if isinstance(entity, ReadyToTest):
+                        yield entity
+                    yield from iterate_ready_to_test_actions(
+                        entity.describe_sub_entities()
+                    )
+                    for conditional_sub_entity in entity.describe_conditional_sub_entities():
+                        yield from iterate_ready_to_test_actions(
+                            conditional_sub_entity[1]
```

Comment:
shouldn't be:
```python3
for conditional_sub_entity in entity.describe_conditional_sub_entities():
    yield from iterate_ready_to_test_actions(
        [conditional_sub_entity[1]]
    )
```

or better:
```python3
yield from iterate_ready_to_test_actions(
    conditional_sub_entity[1] for conditional_sub_entity in entity.describe_conditional_sub_entities()
)
```

Reply:
@ivanpauno 
I don't think the first one is correct.  [describe_conditional_sub_entities](https://github.com/ros2/launch/blob/master/launch/launch/action.py#L95) returns a a tuple of `(text description, iterable(entity))` so conditional_sub_entity[1] should already be iterable

I've got a test case that hits this path [here](https://github.com/ros2/launch/pull/322/files#diff-6b08971ae572019599efe076a73b7b94R119-R135) so I think it's working correctly, but I may be misunderstanding your concern.

I also added a [new test case](https://github.com/ros2/launch/pull/322/files#diff-6b08971ae572019599efe076a73b7b94R137-R163) that has multiple conditional actions, and requires recusing farther to find the ReadyAction

Reply:
yes, you're right, I read the return type incorrectly.